src/internal_rules.jl

```diff
@@ -447,6 +447,65 @@ function EnzymeRules.reverse(config, func::Const{typeof(\)}, ::Type{RT}, cache,
     return (nothing,nothing)
 end
 
+const EnzymeTriangulars = Union{
+    UpperTriangular,
+    LowerTriangular,
+    UnitUpperTriangular,
+    UnitLowerTriangular
+}
+
+function EnzymeRules.augmented_primal(
+    config,
+    func::Const{typeof(ldiv!)},
+    ::Type{RT},
+    Y::Annotation{YT},
+    A::Annotation{AT},
+    B::Annotation{BT}
+) where {RT, YT <: Array, AT <: EnzymeTriangulars, BT <: Array}
+    func.val(Y.val, A.val, B.val)
```

Comment:
Tests presently fail, presumably because you need to cache before running `func.val`

Reply:
Hmm, I thought that the point of caching was to preserve the value before it might be overwritten. So if during the forward pass you do the linear solve and write to Y and that gets overwritten by the subsequent call, in the backward pass you need to retrieve the value from a copy that was not touched by the next function. So this would imply that we need to call the function before making the copy. Is my reasoning off?

Reply:
No you are correct, the code when I commented ran the in place linear solve, then cached -- meaning you would cache the version overwritten by the linear solve. After making the comment, I subsequently fixed the code.

Reply:
But then my logic _is_ backwards. So the cache in the `reverse` call will be retrieved from the next function? Because the rule needs the `Y` after the linear solve, but before any overwrite has happened.
Would this not hold for the `ldiv!` rule for Cholesky factorizations as well?
In any case we still have failures.